shell/renderer/electron_renderer_client.cc

```diff
@@ -141,6 +190,13 @@ void ElectronRendererClient::DidCreateScriptContext(
   gin_helper::Dictionary process_dict(env->isolate(), env->process_object());
   process_dict.SetReadOnly("isMainFrame", render_frame->IsMainFrame());
 
+#if BUILDFLAG(ENABLE_BUILTIN_SPELLCHECKER)
+  base::WeakPtr<SpellCheck> c = GetSpellCheck()->AsWeakPtr();
+
+  process_dict.SetMethod("isWordMisspelled", base::BindRepeating(&IsWordMisspelled, c));
```

Comment:
Hm, so this definitely shouldn't be on `process`.  My guess is it should probably be on something like `webFrame` 🤔

Reply:
Yeah agreed. I'm debating whether to put it on `webFrame` or adding its own module, something like `require('electron').spellcheck`.

Here's what I wrote in the PR description:

> In practice, I have not found a way to pass the `WeakPtr<Spellcheck>` to [the renderer API stack](https://github.com/electron/electron/blob/master/lib/renderer/api/module-list.ts). I was also unable to find a reference back to `ElectronRendererClient`/`RenderClientBase` from an API module.
> If someone with experience can give me some tips on how to achieve it that would be great!

Reply:
@lishid You can find some examples on adding new modules under `shell/renderer/api/` folder. 

Adding the APIs to `webFrame` would be simplest and you only need to move the code to `shell/renderer/api/electron_api_web_frame.cc`.

Reply:
@zcbenz Thank you for the help! In my working copy I was already able to move the code & API to `webFrame`, however, I was unable to find a reference to [the spellchecker object](https://github.com/electron/electron/blob/master/shell/renderer/renderer_client_base.h#L84) which is located on RefererClientBase. Any tips? It seems that all the samples I've seen only references `content::RenderFrame` gotten from a [v8::Context](https://github.com/electron/electron/blob/master/shell/renderer/api/electron_api_web_frame.cc#L93)

Reply:
Since the `RendererClientBase` is guaranteed to have only one instance per process, I think you can add a `RendererClientBase::Get()` method to return the global instance, like what we do in `ElectronBrowserMainParts::Get`.

Chromium also does a similar thing for `ChromeExtensionsRendererClient`, so this pattern should be fine.

Reply:
Great having some code I can reference makes this a lot easier. Will update the PR soon, thanks again!